src/node.cc

```diff
@@ -4412,10 +4386,23 @@ static void StartNodeInstance(void* arg) {
     Isolate::Scope isolate_scope(isolate);
     HandleScope handle_scope(isolate);
     Local<Context> context = Context::New(isolate);
-    Environment* env = CreateEnvironment(isolate, context, instance_data);
-    array_buffer_allocator->set_env(env);
     Context::Scope context_scope(context);
 
+    // FIXME(bnoordhuis) Work around V8 bug: v8::Private::ForApi() dereferences
+    // a nullptr when a context hasn't been entered first.  The symbol registry
+    // is a lazily created JS object but object instantiation does not work
+    // without a context.
+    IsolateData isolate_data(isolate, instance_data->event_loop());
+
+    Environment* env = CreateEnvironment(&isolate_data,
```

Comment:
I confess I'm not sure what you mean by 'direct initialization'.  Making Environment a member of IsolateData?  There can be more than one Environment per isolate so that won't work.

Or do you mean making `env` a stack-allocated variable?